pandas/_libs/parsers.pyx

```diff
@@ -1229,7 +1238,20 @@ cdef class TextReader:
                 na_count = 0
 
             if result is not None and dtype != 'int64':
-                result = result.astype(dtype)
+                if is_extension_array_dtype(dtype):
+                    try:
```

Comment:
why do you have a try/except here? is theere a test that fails? this is a requirements for EA dtypes.

Reply:
If it was a string, but it shouldn't ever be. Taking that out.